src/Plugin/resource/Field/ResourceFieldCollection.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Plugin\resource\Field\ResourceFieldCollection.
+ */
+
+namespace Drupal\restful\Plugin\resource\Field;
+
+class ResourceFieldCollection implements ResourceFieldCollectionInterface {
```

Comment:
or is intended to be the only class, and it should be decorated per data provider?